lib/TvmClient.js

```diff
@@ -263,14 +285,21 @@ class TvmClient {
     const fullUrl = urlJoin(this.apiUrl, endpoint)
     const cacheKey = createCacheKey(fullUrl, this.ow.namespace, this.ow.auth)
 
-    let creds = await this._getCredentialsFromCacheFile(cacheKey)
+    let creds = this._getCredentialsFromCache(cacheKey)
+    if (!creds) { creds = await this._getCredentialsFromCacheFile(cacheKey) }
     if (!creds) {
       creds = await this._getCredentialsFromTVM(fullUrl)
+      this._cacheCredentials(cacheKey, creds)
       await this._cacheCredentialsToFile(cacheKey, creds)
```

Comment:
Ideally as an additional performance optimization it would be great to reduce the latency by writing to the file in the background, i.e not awaiting, however this might probably bring some additional race conditions, something to keep in mind for later

Reply:
or simpler let the user decide if he wants to cache to file or to memory as a constructor option, wdyt? 
but again let's keep this in mind for later